src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -4477,7 +4477,7 @@ private void AddSynthesizedTupleMembersIfNecessary(MembersAndInitializersBuilder
                         {
                             var fieldSyntax = (FieldDeclarationSyntax)m;
 
-                            _ = fieldSyntax.Declaration.Type.SkipScoped(out _).SkipRef(out RefKind refKind);
+                            _ = fieldSyntax.Declaration.Type.SkipScoped(out _).SkipRef(diagnostics, out RefKind refKind);
```

Comment:
>diagnostics [](http://example.com/codeflow?start=87&length=11)

Does this mean we'll report "ref locals are not supported in C#4" rather than "ref fields are not supported in C#4"?

Reply:
Yes.  The same as when thsi was being reported in the parser.  There was no knowledge at the point of time of parsing what context one was in, so you always got the same error about 'ref'.